cristin-biobank/src/test/java/no/unit/nva/cristin/biobank/fetch/FetchBiobankHandlerTest.java

```diff
@@ -0,0 +1,95 @@
+public class FetchBiobankHandlerTest {
+
+    private static final String EXISTING_BIOBANK_IDENTIFIER = "2510604";
+    private final Context context = new FakeContext();
+    private FetchBiobankHandler handlerUnderTest;
+    private ByteArrayOutputStream output;
+    private CristinBiobankApiClient cristinClient = new CristinBiobankClientStub();
+    @BeforeEach
+    public void beforeEach(WireMockRuntimeInfo runtimeInfo) {
+        this.handlerUnderTest = new FetchBiobankHandler(cristinClient, new Environment());
+        this.output = new ByteArrayOutputStream();
+
+    }
+
+    @ParameterizedTest(name = "Should fetch biobank with Id {0} ")
+    @ValueSource(strings = {"2510604", "643747"})
+    public void shouldReturnBiobankWhenExists(String biobankId) throws IOException {
+        var input = new HandlerRequestBuilder<Void>(dtoObjectMapper)
+                        .withPathParameters(Map.of(IDENTIFIER, biobankId))
+                        .withHeaders(Map.of("Accept", "application/ld+json"))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Biobank.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+
+        var biobank = response.getBodyObject(Biobank.class);
+        var expectedId = URI.create("https://api.dev.nva.aws.unit.no/cristin/biobank/" + biobankId);
+
+        assertThat(biobank.getId(), is(equalTo(expectedId)));
+    }
+
+    @ParameterizedTest(name = "Should fail fetch biobank with Id {0} ")
+    @ValueSource(strings = {"133124"})
+    public void shouldReturnFailureWhenBiobankIdDoNotExists(String biobankId) throws IOException {
+        var input = new HandlerRequestBuilder<Void>(dtoObjectMapper)
+                        .withPathParameters(Map.of(IDENTIFIER, biobankId))
+                        .withHeaders(Map.of("Accept", "application/ld+json"))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Biobank.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_GATEWAY)));
+
+    }
+
+    @ParameterizedTest(name = "Should fail fetch biobank with Id {0} ")
+    @ValueSource(strings = { "03747"})
+    public void shouldReturnFailureWhenBiobankIdIsInvalid(String biobankId) throws IOException {
+        var input = new HandlerRequestBuilder<Void>(dtoObjectMapper)
+                        .withPathParameters(Map.of(IDENTIFIER, biobankId))
+                        .withHeaders(Map.of("Accept", "application/ld+json"))
+                        .build();
+
+        handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Biobank.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
```

Comment:
what is an invalid identifier? Looks like a number 2 me? is it because of first digit being zero? what about just using some text?

Reply:
correct, 0 first, is no number